### PR TITLE
feat: un-static FakeDatabase so that it can initialised with AppSettings config

### PR DESCRIPTION
**Pull request by civsiv**
Refactor to improve use of FakeDatabase to make it easier to use with DI, and also clearer when it is being instantiated. Also allows configuration to be passed into it

### CODE REVIEW

File: Examples/BookingSystem.AspNetCore.IdentityServer/Custom/Users/Extensions.cs [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+using OpenActive.FakeDatabase.NET;
 
 namespace IdentityServer
 {
     public static class CustomIdentityServerBuilderExtensions
     {
         public static IIdentityServerBuilder AddFakeUserStore(this IIdentityServerBuilder builder, string jsonLdIdBaseUrl)
         {
-            builder.Services.AddSingleton<IUserRepository>(repo => new UserRepository(jsonLdIdBaseUrl));
+            builder.Services.AddSingleton<IUserRepository>(repo => new UserRepository(jsonLdIdBaseUrl, repo.GetRequiredService<FakeBookingSystem>()));
```
Comment (civsiv) [inline Examples/BookingSystem.AspNetCore.IdentityServer/Custom/Users/Extensions.cs:11]: THis is the concern I have, is this the right way of doing this?

Comment (nickevansuk) [reply Examples/BookingSystem.AspNetCore.IdentityServer/Custom/Users/Extensions.cs:11]: I remember something about `GetRequiredService` not being the ideal thing, but need to check

File: Examples/BookingSystem.AspNetCore/Stores/OrderStore.cs [hunk 2]
```diff
@@ -35,7 +37,7 @@ public override async Task<bool> CustomerCancelOrderItems(OrderIdComponents orde
         {
             try
             {
-                return await FakeBookingSystem.Database.CancelOrderItems(
+                return await _fakeBookingSystem.database.CancelOrderItems(
```
Comment (nickevansuk) [inline Examples/BookingSystem.AspNetCore/Stores/OrderStore.cs:40]: .NET convention is properties start with upper case

```suggestion
                return await _fakeBookingSystem.Database.CancelOrderItems(
```

File: Fakes/OpenActive.FakeDatabase.NET/FakeBookingSystem.cs [hunk 3]
```diff
@@ -19,34 +19,28 @@ namespace OpenActive.FakeDatabase.NET
...
+        {
+            database = FakeDatabase.GetPrepopulatedFakeDatabase().Result;
+
+        }
         /// <summary>
         /// The Database is created as static, to simulate the persistence of a real database
         /// </summary>
-        public static FakeDatabase Database { get; } = FakeDatabase.GetPrepopulatedFakeDatabase().Result;
-
-        public static void Initialise()
```
[the remark this answers is not in the document]
Comment (nickevansuk) [reply Fakes/OpenActive.FakeDatabase.NET/FakeBookingSystem.cs:29]: On reflection, good to keep this and put it wherever DI expects it to go (maybe let's look at what Legend do?)

**Review by nickevansuk — CHANGES_REQUESTED** [review]

File: Fakes/OpenActive.FakeDatabase.NET/FakeBookingSystem.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickevansuk) [reply Fakes/OpenActive.FakeDatabase.NET/FakeBookingSystem.cs:29]: Likely this is in the extension that adds FakeBookingSystem to DI

[the remark this answers is not in the document]
Comment (civsiv) [reply Fakes/OpenActive.FakeDatabase.NET/FakeBookingSystem.cs:29]: @nickevansuk I don't think this is needed still as the Database tables are initialised when FakeBookingSystem is constructed. 

The `Initialise()` method called an arbitrary table to make sure everything was initialised, but that is done in `GetPrepopulatedFakeDatabase()` in the constructor of `FakeBookingSystem`

[the remark this answers is not in the document]
Comment (civsiv) [reply Fakes/OpenActive.FakeDatabase.NET/FakeBookingSystem.cs:29]: Plus `BookingPartnerTable.Get().Wait();` doesn't exist anymore as we've moved `.Get()` to the FakeDatabase.

**Review by nickevansuk — APPROVED** [review]
Looks good!

**Pull request merged**